lookup_plugins: loop over hashes

sl4mmy opened the pull request:
Create a lookup plugin named dict that can be used to loop over hashes.
It converts a dict into a list of key-value pairs, with attributes named
"key" and "value."  Also adds a brief explanation and simple example to
the docs.

Signed-off-by: Kent R. Spillner kspillner@acm.org

sl4mmy committed:
lookup_plugins: loop over hashes

Create a lookup plugin named dict that can be used to loop over hashes.
It converts a dict into a list of key-value pairs, with attributes named
"key" and "value."  Also adds a brief explanation and simple example to
the docs.

Signed-off-by: Kent R. Spillner <kspillner@acm.org>

mpdehaan commented:
Thanks for this.

Can you add  the "Version_added" attribute to the docs so folks will know where this is new?  It also seems to add it after the "with_fileglob" anchor tag, so this will need it's own anchor tag and you will need to move the "with_fileglob" tag down so it's next to the fileglob section.

If that doesn't quite make sense we can adapt, but if you can tweak, we'd appreciate it.  Thank you!

sl4mmy commented:
@mpdehaan D'oh, sorry about that!  I completely missed the anchor tags when adding the original blurb.

I've updated the docs based on your comments, and rebased on latest changes.  Thanks for reviewing!

sl4mmy commented:
@mpdehaan @risaacson _bump_

risaacson commented:
I am working on testing this and so far I am getting stack traces. I'll dig into this further but so far I have the following artifacts.

In my global_vars:

```
users:
  alice:
    name: Alice
    id: 12345
  luna:
    name: Luna
    id: 23456
```

The task from my playbook:

```
- name: Print records.
  debug: msg="User {{ item.key }} is {{ item.value.name }}  {{ item.name.id }}"
  with_dict: users
```

Finally the stacktrace that I am getting:

```
fatal: [test-centos6.recycledthoughts.org] => Traceback (most recent call last):
  File "/Users/ri/repos/github.com/risaacson/ansible/lib/ansible/runner/__init__.py", line 525, in _executor
    exec_rc = self._executor_internal(host, new_stdin)
  File "/Users/ri/repos/github.com/risaacson/ansible/lib/ansible/runner/__init__.py", line 592, in _executor_internal
    items = utils.plugins.lookup_loader.get(items_plugin, runner=self, basedir=basedir).run(items_terms, inject=inject)
  File "/Users/ri/repos/github.com/risaacson/ansible/lib/ansible/utils/plugins.py", line 176, in get
    self._module_cache[path] = imp.load_source('.'.join([self.package, name]), path)
  File "/Users/ri/repos/github.com/risaacson/ansible/lib/ansible/runner/lookup_plugins/dict.py", line 25
    return.append({'key': key, 'value': terms[key]})
          ^
SyntaxError: invalid syntax
```

sl4mmy commented:
@risaacson D'oh!  That's a typo, sorry!  1 sec...

sl4mmy commented:
@risaacson I just rebased on latest changes and squashed in a fix (it should have been ret.append, not return.append).  Again, I'm really sorry about that!  I'm actually running this in production as a local lookup plugin and when I copied it into Ansible's source tree I must have botched a search&replace along the way.
